### ISSUE TITLE
Sentence Classification using BERT/ernie 

### ISSUE THREAD

**Issue by AyanSinhaMahapatra** [nexB/scancode-toolkit#2181]
Ernie -  https://github.com/labteral/ernie

Use Cases - 

1. Determining License Class (Text/Notice/Tag/Reference) of Generated Rule Texts
2. Separating False Positives from License Tags.

Steps - 

1. Fine Tune BERT models on License Texts
2. Train Models on Scancode Rules (For Determining License Class)
3. Train Models on Scancode Rules (Separating False Positives in License Tags)
4. Model Validation and Comparing HyperParameters
5. Reviewing Results

**Comment by AyanSinhaMahapatra** [comment]
Preliminary Results on sentence classification with BERT/ERNIE

1. Determining License Class (Text/Notice/Tag/Reference) of Generated Rule Texts

   - Model - BertBaseUncased (Weights 0.5 GB) 
   - Sentence Length - 8
   - Labels - 2 (False Positive/License Tag)


After 4 Epochs of Fine-Tuning with learning rate 2e-5

   - accuracy on the training data (90%): 0.9948 
   - accuracy on the validation data (10%): 0.9479


2. Separating False Positives from License Tags.


   - Model - BertBaseUncased (Weights 0.5 GB) 
   - Sentence Length - 8
   - Labels - 4 (License Text/Notice/Tag/Reference)


After 12 Epochs of Fine-Tuning with learning rate 2e-5

   - accuracy on the training data (90%): 0.9604 
   - accuracy on the validation data (10%): 0.8398

More tasks - 
 
- hyperparameter tuning
- try bigger models (BertLargeUncased)
- See if Cased Models make any difference (Cased Models differentiate between Uppercase and Lowercase Characters)
- Data Augmentation
- Better Validation Schemes

**Comment by AyanSinhaMahapatra** [comment]
@pombredanne @majurg Update. I will create a PR with the Code today.
